libs/agno/agno/tools/whatsapp.py

```diff
@@ -0,0 +1,212 @@
+import json
+import os
+from typing import Any, Dict, List, Optional
+
+import httpx
+from dotenv import load_dotenv
+
+from agno.tools import Toolkit
+from agno.utils.log import logger
+
+# Try to load from both .env and .envrc
+load_dotenv()
+
+
+class WhatsAppTools(Toolkit):
+    """WhatsApp Business API toolkit for sending messages."""
+
+    base_url = "https://graph.facebook.com"
+
+    def __init__(
+        self,
+        access_token: Optional[str] = None,
+        phone_number_id: Optional[str] = None,
+        version: str = "v22.0",
+        recipient_waid: Optional[str] = None,
+    ):
+        """Initialize WhatsApp toolkit.
+
+        Args:
+            access_token: WhatsApp Business API access token
+            phone_number_id: WhatsApp Business Account phone number ID
+            version: API version to use
+            recipient_waid: Default recipient WhatsApp ID (optional)
+        """
+        super().__init__(name="whatsapp")
+
+        # Core credentials
+        self.access_token = access_token or os.getenv("WHATSAPP_ACCESS_TOKEN") or os.getenv("ACCESS_TOKEN")
+        if not self.access_token:
+            logger.error("WHATSAPP_ACCESS_TOKEN not set. Please set the WHATSAPP_ACCESS_TOKEN environment variable.")
+
+        self.phone_number_id = phone_number_id or os.getenv("WHATSAPP_PHONE_NUMBER_ID") or os.getenv("PHONE_NUMBER_ID")
+        if not self.phone_number_id:
+            logger.error(
+                "WHATSAPP_PHONE_NUMBER_ID not set. Please set the WHATSAPP_PHONE_NUMBER_ID environment variable."
+            )
+
+        # Optional default recipient
+        self.default_recipient = recipient_waid or os.getenv("WHATSAPP_RECIPIENT_WAID") or os.getenv("RECIPIENT_WAID")
+
+        # API version
+        self.version = version or os.getenv("WHATSAPP_VERSION") or os.getenv("VERSION", "v22.0")
+
+        # Register methods that can be used by the agent
+        self.register(self.send_text_message_sync)
+        self.register(self.send_template_message_sync)
```

Comment:
What about async? Maybe have a `async_mode:bool` attribute at the top? We don't really build async endpoints yet, but that could be a good pattern

Reply:
added